mino/minogrpc/overlay.go

```diff
@@ -177,18 +184,30 @@ func (o overlayService) Stream(stream Overlay_StreamServer) error {
 		newCtx := stream.Context()
 		newCtx = metadata.NewOutgoingContext(newCtx, header)
 
-		clientStream, err := cl.Stream(newCtx)
+		cs, err := cl.Stream(newCtx)
 		if err != nil {
 			err = xerrors.Errorf("failed to get stream for client '%s': %v",
 				addr.String(), err)
 			fabric.Logger.Err(err).Send()
 			return err
 		}
 
-		sender.participants[addr.String()] = clientStream
+		clientStream := newSafeOverlayStream(cs)
+
+		sender.participants.Store(addr.String(), clientStream)
 
 		// Sending the routing info as first messages to our children
-		clientStream.Send(&OverlayMsg{Message: overlayMsg.Message})
+		clientStream.Send(&Envelope{
+			From:         o.addr.String(),
+			PhysicalFrom: o.addr.String(),
```

Comment:
I would use something more meaningful like `LastHop` or `Relay`. Now, is it really necessary ? You're supposed to know the mesh.

Reply:
IMO it can't be more explicit than that. `LastHop` is already speaking at a logical level, where here I want to know who sent me this message, regardless its content.
It is hard to know for a node who sent a message because the listen function receive messages from the orchestrator and the children of a node.